### DISCUSSION TITLE
"One" digit can't be recognized

### DISCUSSION THREAD

**Discussion by sshalyminov** [jomjol/AI-on-the-edge-device#1375] [General]
### The Problem

I use this great tool on my digital electricity counter. After tuning ROIs position and size it recognise all digits excellent except "1" - it is recognized as "N" or "7". Here is example:
![image](https://user-images.githubusercontent.com/25504725/196168607-0178c866-3616-4006-bbc7-b4727b6f237d.png)
Original image from Recognize page:
![total_dig6](https://user-images.githubusercontent.com/25504725/196168682-1e0f192e-e26a-45e9-92af-159d8b6becb3.jpg)

Digits are a bit angled because of counter screen composition, everything is aligned, all ROIs have same Y coordinate.

Latest release 12.0.1 firmware is installed and dig-class11_1430_s2.tflite is used from this latest release.

As I wrote earlier all digits are recognized perfectly, only "1" has problem. Is this possible to fix?

### Version

12.0.1

**Comment by jomjol** [comment]
This is a topic of the neural network itself. The contrast is limited and right above the "1" there is another line (the end of the LCD display).
A "1" with a bar on top looks for the system like a "7". You could send me some training data of your device, that might help.

**Comment by sshalyminov** [comment]
Can you please give me exact instructions what you need? Sorry for noob question.
If I understand right I need to save wrongly recognized numbers pictures from Recognition page and name it... how?
"Naming: x_ARBITRARY.jpg, where x = value 0 ... 9 + N" - I'm not sure I understand this right way.

**Comment by mbrammer** [comment]
I have a similar problem. I also have the 7 detected as 1. Is there a way to help the AI  to learn? So I could say "No, this is a 1"?

![image](https://user-images.githubusercontent.com/2786547/197408557-3dfed78f-fd6c-4b94-9b53-0a0b3af9894b.png)

**Comment by DonKracho** [comment]
On the 3rd to 5th digit you can see a reflection of light. Looks like a fluorescent tube. I think it is nearly impossible to train the digit detection that top line to be a reflection.  Avoid these light reflections, and most likely everything will be fine.

**Comment by sshalyminov** [comment]
Can anyone help me? What required for neural network training? I had read instruction from https://github.com/jomjol/AI-on-the-edge-device/wiki/Neural-Network-Types but it's not clear...

**Comment by ghost** [comment]
You need images. https://github.com/haverland/collectmeterdigits describes how to do. After you labeled the images put them in https://github.com/haverland/Tenth-of-step-of-a-meter-digit/tree/master/images. Best in /images/collected/<typeofmeter>/<ashortcutofyou>/

This you can later share as pull request.

Run the notebook [dig-class100-s2.ipynb](https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/master/dig-class100-s2.ipynb) or [dig_class100-s1.ipynb](https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/master/dig_class100-s1.ipynb) s1 is bigger ans slower. s2 is in most cases the best.

With [Compare_all_tflite.ipynb](https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/master/Compare_all_tflite.ipynb) you can compare it against the other models.

You model (.tflite) is in output folder with a timestamp.

EDIT: Before you start collecting images, please setup your ROIs like described in collectmeterdigits. Its easier to label later and has a better accuracy in recognition. (Uncheck "lock aspect ratio")

**Comment by mbrammer** [comment]
> On the 3rd to 5th digit you can see a reflection of light. Looks like a fluorescent tube. I think it is nearly impossible to train the digit detection that top line to be a reflection. Avoid these light reflections, and most likely everything will be fine.

I repositioned the camera now and the reflection is gone. From my perspective the digit recognition should now work flawless, but the software is a different opinion.

![image](https://user-images.githubusercontent.com/2786547/199454535-af09e9b0-fb55-4dc8-9c58-52156f474cf4.png)

The fourth digit is a 3, but recognized as 1 o.O

**Comment by jomjol** [comment]
The settings are still not like recommended! The ROIs are much too big. Plese follow the recommendations on this wiki page: https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration

The outer border should be 20%

**Comment by DonKracho** [comment]
B.t.w. I had to uncheck the "Lock aspect ratio" box to get the digits match tight to the real numbers and then entered x and y size manually to be the same dimensions for all digits. [sample image](https://www.thingiverse.com/thing:5595270) The contrast isn't perfect either.
